.travis.yml

```diff
@@ -14,7 +14,19 @@ script:
   - yarn lint
   - yarn tsc
   - yarn test
+  if: branch = master
+  - yarn build:docs
 before_install: yarn global add greenkeeper-lockfile@2
 before_script: greenkeeper-lockfile-update
 after_script: greenkeeper-lockfile-upload
 after_success: ./deploy.sh
+deploy:
```

Comment:
I think you should use [these](https://docs.travis-ci.com/user/build-stages/#specifying-stage-order-and-conditions) conditions

Reply:
@fenilkanjani check this out, see if it makes sense. These are the specific instructions for deploying to github pages.

https://docs.travis-ci.com/user/deployment/pages/

The options you shared are for defining a custom deploy stage, here we're passing options to the `github pages` provider itself, which is taking care of the deployment

Reply:
What I meant was, should not we run this conditionally with:
```
stages:
  - compile
  - test
  - name: deploy
    if: branch = master
```

Reply:
The conditional behaviour is already there, it's specified at the line below. Let's discuss this in-person after standup?

https://github.com/action-land/action-land/pull/25/files#diff-354f30a63fb0907d4ad57269548329e3R31

Reply:
- As long as our build is a multi-job (unlike PWA), then it's fine to just use the `deploy` hook to update documentation.

@fenilkanjani Documentation for the `on` flag in the `deploy` life-cycle method.

> When all conditions specified in the on: section are met, your build will deploy.

https://docs.travis-ci.com/user/deployment/#conditional-releases-with-on